mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/MessageIdMapperTest.java

```diff
@@ -773,6 +773,67 @@ public void countUnseenMessageShouldTakeCareOfMessagesMarkedAsUnread() throws Ex
     }
 
     @ContractTest
+    public void setFlagsShouldNotUpdateModSeqWhenNoop() throws Exception {
+        message1.setUid(mapperProvider.generateMessageUid());
+        long modSeq = mapperProvider.generateModSeq(benwaInboxMailbox);
+        message1.setModSeq(modSeq);
+        message1.setFlags(new Flags(Flag.SEEN));
+        sut.save(message1);
+
+        sut.setFlags(message1.getMessageId(),
+            ImmutableList.of(message1.getMailboxId()),
+            new Flags(Flag.SEEN),
+            FlagsUpdateMode.ADD);
+
+        List<MailboxMessage> messages = sut.find(ImmutableList.of(message1.getMessageId()), MessageMapper.FetchType.Body);
+        assertThat(messages).hasSize(1);
+        assertThat(messages.get(0).getModSeq()).isEqualTo(modSeq);
+    }
+
+    @ContractTest
+    public void setFlagsShouldNotUpdateFlagsWhenNoop() throws Exception {
+        message1.setUid(mapperProvider.generateMessageUid());
+        long modSeq = mapperProvider.generateModSeq(benwaInboxMailbox);
+        message1.setModSeq(modSeq);
+        Flags flags = new Flags(Flag.SEEN);
+        message1.setFlags(flags);
+        sut.save(message1);
+
+        sut.setFlags(message1.getMessageId(),
+            ImmutableList.of(message1.getMailboxId()),
+            flags,
+            FlagsUpdateMode.ADD);
+
+        List<MailboxMessage> messages = sut.find(ImmutableList.of(message1.getMessageId()), MessageMapper.FetchType.Body);
```

Comment:
IMO, if I understand correctly you test does not test nothing. Ok you test that if I take a message A with flag F and I ask the message mapper to add to A flag F, when I retrieve the message after, the set of flag of A is the same now than before.
But this does not test that MessageIdMapper doesn't really not perform an update..
So if you agree with what I said either the name of test should change to something like 'addingFlagToAMessageThatAlreadyHasThisFlagShouldResultInNoChange' or either the code of this test should change